### PR TITLE
Added XGBoost, LGBM, SVR algorithms

### LINKED ISSUE #45: Add XGBoost, LGBM, SVR algorithms
### Prerequisites

- [X] I am an Open Source Contributor
- [X] I checked to make sure that this issue has not already been filed

### Description

I would like to implement XGBoost, LGBM, SVM algorithms for above dataset.

### Screenshots

_No response_

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/Rakesh9100/ML-Project-Drug-Review-Dataset/blob/main/.github/CONTRIBUTING_GUIDELINES.md) of this project.

### PR DESCRIPTION
**Pull request by pranavs1205**
Closes #45 

Title: Addition of XGBoost, LGBM, and SVR Algorithms

# Description👨‍💻 
Title: Addition of XGBoost, LGBM, and SVR Algorithms

This pull request adds XGBoost, LGBM, and SVR algorithms to our machine learning codebase. These algorithms enhance predictive performance, scalability, and versatility. The implementation follows best practices and includes rigorous testing, documentation, and code organization. By merging this pull request, we can leverage the power of these algorithms to deliver more accurate predictions and expand our machine learning capabilities. Feedback and suggestions for improvement are welcome.

# Type of change📄

- [] New feature (non-breaking change which adds functionality)

# How this has been tested✅

The implementation follows best practices and includes rigorous testing, documentation, and code organization

# Checklist✅ 

- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have added demonstration in the form of GIF/video file
- [] I am an Open Source Contributor

# Screenshots/GIF📷

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -6,3 +6,4 @@ numpy==1.24.3
 pandas==2.0.1
 scikit-learn==1.2.2
 seaborn==0.12.2
+XGBOOST
```
Comment (abhisheks008) [inline requirements.txt]: provide the version of xgboost.

**Review by abhisheks008 — CHANGES_REQUESTED** [review]
@pranavs1205

File: requirements.txt (hunk 1 above)
Comment (abhisheks008) [reply requirements.txt]: Can you mention the issue number please?

Comment (pranavs1205) [reply requirements.txt]: 45

**Review by abhisheks008 — APPROVED** [review]
LGTM!

**Comment by Rakesh9100** [comment]
Pls provide the metrics and plots output screenshots and mention the #issuenumber this PR closes @pranavs1205

**Comment by Rakesh9100** [comment]
Pls provide the metrics and plots output screenshots @pranavs1205

**Comment by pranavs1205** [comment]
> Pls provide the metrics and plots output screenshots @pranavs1205
image

**Comment by pranavs1205** [comment]
> Pls provide the metrics and plots output screenshots @pranavs1205

image

**Comment by Rakesh9100** [comment]
Mean squared and R2 score metrics ouputs @pranavs1205

**Comment by pranavs1205** [comment]
LGBM Metrics:
MSE for training:  5.614266291975168
MSE for testing:  6.08444646567784
R2 score for training:  0.4762102373816558
R2 score for testing:  0.4362075610535959

**Comment by Rakesh9100** [comment]
Approved!! Congrats on getting your PR merged to this repo under GSSoC23 🎉 @pranavs1205

**Pull request merged**